import-export-cli/cmd/mg/listAPIs.go

```diff
@@ -0,0 +1,96 @@
+* software distributed under the License is distributed on an
+* "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+* KIND, either express or implied.  See the License for the
+* specific language governing permissions and limitations
+* under the License.
+ */
+
+package mg
+
+import (
+	"encoding/base64"
+	"fmt"
+	"os"
+	"strconv"
+
+	"github.com/spf13/cobra"
+	mgImpl "github.com/wso2/product-apim-tooling/import-export-cli/impl/mg"
+	"github.com/wso2/product-apim-tooling/import-export-cli/utils"
+	"golang.org/x/crypto/ssh/terminal"
+)
+
+var (
+	listApisCmdAPIType string
+	listApisCmdLimit   string
+	listApisUsername   string
+)
+
+const listApisCmdLiteral = "apis"
+const listApisCmdShortDesc = "Display a list of all APIs or a filtered set of APIs"
+const listApisCmdLongDesc = `Display a list of all APIs or filtered by apiType using the flag --type, -t`
+
+var listApisCmdExamples = utils.ProjectName + ` ` + mgCmdLiteral + ` ` + listApisCmdLiteral + `-h https://localhost:9095 -u admin
+ ` + utils.ProjectName + ` ` + mgCmdLiteral + ` ` + listApisCmdLiteral + ` -t http -h https://localhost:9095 -u admin -l 100
+ ` + utils.ProjectName + ` ` + mgCmdLiteral + ` ` + listApisCmdLiteral + ` -t ws -h https://localhost:9095 -u admin`
+
+var mgListAPIsResourcePath = "/apis"
+
+// ListApisCmd represents the apis command
+var ListApisCmd = &cobra.Command{
+	Use:     listApisCmdLiteral,
+	Short:   listApisCmdShortDesc,
+	Long:    listApisCmdLongDesc,
+	Example: listApisCmdExamples,
+	Run: func(cmd *cobra.Command, args []string) {
+		utils.Logln(utils.LogPrefixInfo + listApisCmdLiteral + " called")
+
+		// handle auth
+		fmt.Print("Enter Password: ")
+		password, err := terminal.ReadPassword(0)
+		fmt.Println()
+		if err != nil {
+			utils.HandleErrorAndExit("Error reading password", err)
+		}
+		authToken := base64.StdEncoding.EncodeToString([]byte(listApisUsername + ":" + string(password)))
+
+		//handle parameters
+		if listApisCmdLimit == "" {
+			listApisCmdLimit = strconv.Itoa(utils.DefaultApisDisplayLimit)
+			fmt.Print("Limit flag not set. Set to default: " + listApisCmdLimit + "\n")
+		}
+		queryParams := make(map[string]string)
+		queryParams["limit"] = listApisCmdLimit
+		queryParams["apiType"] = listApisCmdAPIType
+		total, count, apis, err := mgImpl.GetAPIList(authToken,
+			mgwAdapterHost+MgBasepath+mgListAPIsResourcePath,
+			queryParams)
+		if err != nil {
+			utils.HandleErrorAndExit("Error while retrieving or processing received APIs", err)
+		}
+		fmt.Fprintf(os.Stderr, "APIs total: %v received: %v\n", total, count)
+		mgImpl.PrintAPIs(apis)
+	},
+}
+
+func init() {
+	ListCmd.AddCommand(ListApisCmd)
+
+	ListApisCmd.Flags().StringVarP(&mgwAdapterHost, "host", "c", "", "The adapter host url with port")
+	ListApisCmd.Flags().StringVarP(&listApisCmdAPIType, "type", "t", "", "API type to filter the APIs")
+	ListApisCmd.Flags().StringVarP(&listApisCmdLimit, "limit", "l", "", "Maximum number of apis to return")
```

Comment:
Shall we change this as follows?
```suggestion
	ListApisCmd.Flags().StringVarP(&listApisCmdLimit, "limit", "l", "", "Maximum number of APIs to return")
```

Reply:
Thanks for the suggestion. Updating it.